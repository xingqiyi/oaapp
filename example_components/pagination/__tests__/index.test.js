import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Pagination from '../index';

describe('Pagination', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Pagination total={5} current={3} />,
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.am-pagination')).toHaveLength(1);
  });

  it('renders mode="number" correctly', () => {
    const wrapper = mount(
      <Pagination mode="number" total={5} current={3} />,
    );

    expect(wrapper.find('.am-flexbox-item')).toHaveLength(0);
  });
});
