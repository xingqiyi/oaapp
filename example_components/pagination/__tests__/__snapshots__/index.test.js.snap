// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div
  class="am-pagination"
>
  <div
    class="am-flexbox am-flexbox-align-center"
  >
    <div
      class="am-flexbox-item am-pagination-wrap-btn am-pagination-wrap-btn-prev"
    >
      <a
        aria-disabled="false"
        class="am-button am-button-inline"
        role="button"
      >
        <span>
          上一页
        </span>
      </a>
    </div>
    <div
      aria-live="assertive"
      class="am-flexbox-item am-pagination-wrap"
    >
      <span
        class="active"
      >
        3
      </span>
      /
      <span>
        5
      </span>
    </div>
    <div
      class="am-flexbox-item am-pagination-wrap-btn am-pagination-wrap-btn-next"
    >
      <a
        aria-disabled="false"
        class="am-button am-button-inline"
        role="button"
      >
        <span>
          下一页
        </span>
      </a>
    </div>
  </div>
</div>
`;
